docs/src/tutorials/magnetism.ipynb

```diff
@@ -0,0 +1,577 @@
+    "    refl1d_sld_4(100, 2, magnetism=refl1d.names.Magnetism(rhoM=10, thetaM=70)) | \n",
+    "    refl1d_vacuum(0, 2)\n",
+    ") \n",
+    "probe = refl1d.names.QProbe(\n",
+    "        Q=model_coords,\n",
+    "        dQ=np.zeros(len(model_coords)),\n",
+    "        intensity=1,\n",
+    "        background=0,\n",
+    "    )\n",
+    "\n",
+    "four_probes = [probe, None, None]\n",
+    "polarized_probe = refl1d.names.PolarizedQProbe(xs=four_probes, name='polarized')\n",
+    "experiment = refl1d.names.Experiment(probe=polarized_probe, sample=refl1d_sample)\n",
+    "model_data_magnetism_ref1d = experiment.reflectivity()[0][1]\n",
+    "plt.plot(model_coords, model_data_magnetism_ref1d, '-k', label='Refl1d', linewidth=4)\n",
+    "\n",
+    "# EasyReflectometry\n",
+    "interface.switch('refl1d')\n",
+    "model.interface = interface\n",
+    "model_interface = model.interface()\n",
+    "model_interface.magnetism = True\n",
+    "model_interface._wrapper.update_layer(list(model_interface._wrapper.storage['layer'].keys())[1], magnetism_rhoM=10, magnetism_thetaM=70)\n",
+    "model_interface._wrapper.update_layer(list(model_interface._wrapper.storage['layer'].keys())[2], magnetism_rhoM=5, magnetism_thetaM=175)\n",
+    "model_data_magnetism_easy = model.interface().fit_func(\n",
+    "    model_coords,\n",
+    "    model.uid,\n",
+    ")\n",
+    "plt.plot(model_coords, model_data_magnetism_easy, 'r-', label=f'EasyReflect ({model_interface.name})', linewidth=2)\n",
+    "\n",
+    "plot_apply_makeup()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d52efccc",
+   "metadata": {},
+   "source": [
+    "The two models agree when the magnetic parameters are the same."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7033f755",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(max(abs(model_data_magnetism_easy - model_data_magnetism_ref1d)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "971baee6",
+   "metadata": {},
+   "source": [
+    "# Afterthoughts \n",
+    "Just for completion we will do a few additional calculations and comparisons of reflectivity."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7af84a69",
+   "metadata": {},
+   "source": [
+    "## Refl1d polarized probe for a single layer sample\n",
+    " This study is done with magnetism to show the results for polarized probe."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "352c35e9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The magnetism is set to 8. \n",
+    "# This would double (pp) and cancel out (mm) the magnitude of the reflectivity oscillations when its angle is set to 90\n",
+    "# This would the strongest spin-flipping (pm and mp) when its angle is set to 0  \n",
+    "refl1d_sample = (\n",
+    "    refl1d_si(0, 2) | \n",
+    "    refl1d_sld_8(150, 2, magnetism=refl1d.names.Magnetism(rhoM=8, thetaM=45)) |\n",
```

Comment:
Should this be 45 or 90?
Also, setting rho_M to 8 will cancel the refelectivity for one spin state, but not if you also change the angle

Reply:
Changed comment, now states the 90 and 0 degree would give the extreme condition and states that we however set the value to 45 degree